core/src/main/scripts/importer/libImportOncokb.py

```diff
@@ -0,0 +1,236 @@
+            file_handle.close()
+            return first_line
+
+
+def find_meta_file_by_fields(study_dir, field_dict):
+    """In the study directory find the meta file that holds specified field values."""
+    filenames = os.listdir(study_dir)
+    meta_files = [ file for file in filenames if file.startswith('meta_')]
+    meta_files = [ study_dir + "/" + file for file in meta_files ]
+    for meta_file in meta_files:
+        fields = read_meta_file(meta_file)
+        match_found = True
+        for field_name, field_value in field_dict.items():
+            if field_name not in fields or fields[field_name] != field_value:
+                match_found = False
+        if match_found:
+            return os.path.join(study_dir, meta_file)
+    return None
+
+
+def find_data_file_from_meta_file(metafile_path):
+    """From specified meta file return the value of the 'data_filename' field."""
+    fields = read_meta_file(metafile_path)
+    return fields['data_filename']
+
+
+def read_meta_file(metafile_path):
+    """Read fields of specified meta into a dict."""
+    fields = {}
+    with open(metafile_path) as metafile:
+        for line in metafile:
+            if line == '\n':
+                continue
+            match = re.findall(r'\s*(\S+)\s*:\s*(\S+)', line)[0]
+            fields[match[0]] = match[1]
+    return fields
+
+def get_oncokb_cancer_genes():
+    """Do a call to OncoKB to retrieve the Cancer Gene List. """
+    request_url = "https://public.api.oncokb.org/api/v1/utils/allCuratedGenes"
+    request = requests.get(url=request_url)
+
+    if request.ok:
+        return request.json()
+    else:
+        if request.status_code == 404:
+            raise ConnectionError(
+                "An error occurred when trying to connect to OncoKB for retrieving of mutation annotations.")
+        else:
+            request.raise_for_status()
+
+def get_annotated_oncokb_cancer_genes_by_entrezId():
+    """ Get a list with the Entrez Gene IDs of the OncoKB Cancer Gene List, only the genes that have
+        annotations. """
+    cancer_genes = get_oncokb_cancer_genes()
+    entrez_ids = []
+    for gene in cancer_genes:
+        entrez_ids += [gene["entrezGeneId"]]
+    return entrez_ids
+
+def filter_payload(payload_list, sv):
+    """ Remove the genes that are not an OncoKB Cancer Gene List from the payload. """
+    cancer_genes = get_annotated_oncokb_cancer_genes_by_entrezId()
+    filtered_payload = []
+    for element in payload_list:
+        parsed_element = json.loads(element)
+        if sv:
+            if parsed_element["geneA"]["entrezGeneId"] in cancer_genes and parsed_element["geneB"]["entrezGeneId"] in cancer_genes:
+                filtered_payload += [element]
+        else:
+            if parsed_element["gene"]["entrezGeneId"] in cancer_genes:
+                filtered_payload += [element]
+    return filtered_payload
+
+def fetch_oncokb_annotations(payload_list, request_url, sv=False):
```

Comment:
I don't see usage of sv. Are you adding later?

Reply:
We tried to implement Structural Variant functions (at least for the update script) as a "preparation" when we have SV finished. However, this needs to be finalized once the Structural Variants are fully working.